packages/astro/src/core/config/schema.ts

```diff
@@ -675,13 +626,37 @@ export const AstroConfigSchema = z.object({
 						...BUILTIN_PROVIDERS,
 						...(fonts.providers ?? []).map((provider) => provider.name),
 					];
-					for (const family of fonts.families) {
+					const visited = new Map<string, 'name' | 'as'>();
+
+					for (let i = 0; i < fonts.families.length; i++) {
+						const family = fonts.families[i];
+
+						// Check for invalid providers
 						if (!providersNames.includes(family.provider)) {
 							ctx.addIssue({
 								code: z.ZodIssueCode.custom,
 								message: `Invalid provider "${family.provider}". Please use of the following: ${providersNames.map((name) => `"${name}"`).join(', ')}`,
+								path: ['families', i, 'provider'],
+							});
+						}
+
+						// Check for name/as conflicts
+						const name = getFamilyName(family);
+						const key = 'as' in family ? 'as' : 'name';
+						const existing = visited.get(name);
+						if (existing) {
+							ctx.addIssue({
+								code: z.ZodIssueCode.custom,
+								message:
+									key === 'name' && existing === 'name'
+										? `2 families have the same **name** property: "${name}". Names have to be unique, you can override one of these family name by specifying the **as** property. Read more at TODO:`
```

Comment:
Could it be more than 2? Should it say "multiple"?

Reply:
Yes it could be more than 2 but we fail as soon as there are 2. I'll update the wording